### PR TITLE
lavfi/dnn/dnn_backend_tf.h: Add Documentation

### PR DESCRIPTION
**Pull request by shubhanshu02**
## Description

Add documentation to `dnn_backend_tf.h` in `libavfilter/dnn`

@guoyejun, please review these changes. 

Also, I wanted to ask should I document the functions in `dnn_backend_tf.c`, which are not directly accessible outside the module?

### CODE REVIEW

**Comment by guoyejun** [comment]
i'm not sure about it, please refer to other internal functions.

**Comment by shubhanshu02** [comment]
I tried looking into files in folders other than `libavfilter`. There was documentation in the header file in most of them, while for the functions in the `.c` file (which were not mentioned in the `.h` file), I couldn't find any docstrings. Some of them had inline comments within those functions, though.

File: libavfilter/dnn/dnn_backend_tf.h [hunk 1]
```diff
@@ -29,11 +29,57 @@
...
 
+/**
+ * @brief Execute TensorFlow model in synchronous mode.
+ *
+ * It parses the data from the input frame, preprocesses
+ * the input frame data if required and fills the input
+ * Tensor. Then it run the TensorFlow session and fills
+ * the output after post processing it if required.
+ *
+ * Currently only single output is supported.
```
Comment (guoyejun) [inline libavfilter/dnn/dnn_backend_tf.h:58]: maybe we can pending this after Fu Ting's patch is merged, he added multiple outputs for tf backend.

Comment (shubhanshu02) [reply libavfilter/dnn/dnn_backend_tf.h:58]: Sure. Do you mean - this function's documentation or the line "Currently only single output is supported"?

Comment (guoyejun) [reply libavfilter/dnn/dnn_backend_tf.h:58]: yes

**Pull request closed without merging**